src/dev/Web/EastSeat.ResourceIdea.Web/Components/Pages/WorkItems/AddWorkItem.razor.cs

```diff
@@ -0,0 +1,296 @@
+            else if (EngagementId.HasValue)
+            {
+                NavigationSource = "engagement";
+            }
+            else
+            {
+                NavigationSource = "workitems";
+            }
+        }
+
+        await LoadEngagements();
+        await LoadEmployees();
+        
+        if (IsEngagementPreSelected)
+        {
+            SelectedEngagementId = EngagementId!.Value.ToString();
+        }
+        
+        IsLoading = false;
+        StateHasChanged();
+    }
+
+    private async Task LoadEngagements()
+    {
+        try
+        {
+            if (ClientId.HasValue)
+            {
+                // Load engagements for specific client
+                var clientQuery = new GetEngagementsByClientQuery(1, 100)
+                {
+                    ClientId = EastSeat.ResourceIdea.Domain.Clients.ValueObjects.ClientId.Create(ClientId.Value)
+                };
+                clientQuery.TenantId = ResourceIdeaRequestContext.Tenant;
+
+                var response = await Mediator.Send(clientQuery);
+                if (response.IsSuccess && response.Content.HasValue)
+                {
+                    Engagements = response.Content.Value;
+                }
+                else
+                {
+                    HasError = true;
+                    ErrorMessage = "Failed to load engagements for the selected client. Please try again later.";
+                }
+            }
+            else
+            {
+                // Load all engagements for the tenant
+                var query = new GetAllEngagementsQuery(1, 100);
+                query.TenantId = ResourceIdeaRequestContext.Tenant;
+
+                var response = await Mediator.Send(query);
+                if (response.IsSuccess && response.Content.HasValue)
+                {
+                    Engagements = response.Content.Value;
+                }
+                else
+                {
+                    HasError = true;
+                    ErrorMessage = "Failed to load engagements. Please try again later.";
+                }
+            }
+        }
+        catch (InvalidOperationException ex)
+        {
+            HasError = true;
+            ErrorMessage = "An error occurred while loading engagements. Please try again later.";
+            Console.Error.WriteLine($"InvalidOperationException in LoadEngagements: {ex}");
+        }
+        catch (HttpRequestException ex)
+        {
+            HasError = true;
+            ErrorMessage = "A network error occurred while loading engagements. Please check your connection and try again.";
+            Console.Error.WriteLine($"HttpRequestException in LoadEngagements: {ex}");
+        }
+        catch (TimeoutException ex)
+        {
+            HasError = true;
+            ErrorMessage = "The request timed out while loading engagements. Please try again later.";
+            Console.Error.WriteLine($"TimeoutException in LoadEngagements: {ex}");
```

Comment:
I am going to setup logging using Azure Application Insights, don't use `Console.Error.WriteLine`.